src/middleware/authenticateToken.ts

```diff
@@ -0,0 +1,58 @@
+// authenticateToken.ts
+
+import jwt from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import properties from '../properties';
+import { AuthenticatedUser } from '../types/AuthenticatedUser';
+import logger from '../logger';
+
+const SECRET_KEY = properties.secretKey;
+
+export function authenticateToken(req: Request, res: Response, next: NextFunction) {
+
+  try {
+    const token = req.headers['authorization'];
+    if (typeof token === 'string') {
+      const tokenString = token.split(' ')[1]; 
+      const user = jwt.verify(tokenString, SECRET_KEY);
+      req.user = user; 
+      next();
+    } else {
+      return res.status(401).json({
+        success: false,
+        error: {
+          error: true,
+          message: 'Unauthorized',
+          path: req.baseUrl,
+        },
+      })
+    } 
+  } catch (err) {
+    logger.error(`Error occured in authenticateToken : ${err}`);
+    return res.status(403).json({
+      success: false,
+      error: {
+        error: true,
+        message: String(err),
```

Comment:
Do not send the err message, send a generic message like `Username/Email or Password is incorrect`